rd_ui/app/scripts/controllers/query_view.js

```diff
@@ -96,7 +96,10 @@
 
     $scope.saveQuery = function(options, data) {
       if (data) {
-        data.id = $scope.query.id;
+        if ($scope.query.id)
+          data.id = $scope.query.id;
+        else // this is a new query, pass initial params (like datasource) and data (like name)
+          data = _.extend($scope.query, data)
```

Comment:
We use parentheses even for single line `if` statements.

But maybe just avoid saving (i.e. exit `saveQuery`) if `query.id` is `null`?